File: 02 백준/01 자료구조/Python/Prob4358.py

"""
title  : 4358. 생태학 (Silver 2)
time   : 344ms       408ms
memory : 31120KB     112016KB
"""

import sys

def solution():

    input = sys.stdin.readline

    trees = {}
    total_tree = 0

    while True:

        tree = input().rstrip()

        if not tree:
            break

        trees[tree] = trees.get(tree, 0) + 1
        total_tree += 1
    
    for key in sorted(trees.keys()):
        print("{0} {1:.4f}".format(key, (trees.get(key) / total_tree) * 100))

solution()

File: 02 백준/01 자료구조/Java/src/Prob4358.java

/**
* title  : 4358. 생태학 (Silver 2)
* time   : ms
* memory : KB
*/

import java.io.*;
import java.util.*;

public class Prob4358 {

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        HashMap<String, Integer> treeMap = new HashMap<>();
        
        String name = "";
        int allCount = 0;

        while ((name = br.readLine()) != null) {
            treeMap.put(name, treeMap.getOrDefault(name, 0) + 1);
            allCount++;
        }

        System.out.println(allCount);

    }
}